### PR TITLE
Fix string|float-to-int convertion errors

### PR DESCRIPTION
**Pull request by jchaffraix**
There were 3 major errors:
- int fields from the DB that are returned as string by mysqli.
- date('s|d|y|U') that returns an int as a string.
- PHP float truncating functions that returns a float.

_Update by cpeel_: Sandbox avail at https://www.pgdp.org/~cpeel/c.branch/julien_fix_int_to_str_conv/

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -24,7 +24,7 @@ parameters:
           path: pinc/3rdparty/mediawiki/DairikiDiff.php
         - message: '#Comparison operation .* results in an error#'
           path: pinc/3rdparty/mediawiki/DiffEngine.php
-        - message: '#^Parameter \#2 \$num of function array_fill expects int, float given\.$#'
+        - message: '#^Parameter \#2 \$.* of function array_fill expects int, float given\.$#'
```
Comment (jchaffraix) [inline phpstan.neon:27]: That's a small mistake I made in the previous PR that makes PHPStan fails: PHP 8 does actually see the error but with a different parameter name.

File: pinc/Pool.inc [hunk 2]
```diff
@@ -35,17 +35,18 @@ class Pool extends Stage
...
-        $access_minima,
-        $after_satisfying_minima,
-        $evaluation_criteria,
-        $access_change_callback,
-        $description,
-        $document,
-        $foo_Header,
-        $foo_field_name,
-        $blather
+        string $id,
```
Comment (jchaffraix) [inline pinc/Pool.inc:38]: I added this type to prove that that the `$id` parameter is a string and thus we need to manually ignore some issue in PHPStan on `pinc/stages.inc`.

The other types in the constructors were added bonuses.

All the types are copied from the class member's.

**Review by cpeel — APPROVED** [review]

File: pinc/stages.inc [hunk 3]
```diff
@@ -307,7 +307,7 @@ new Stage(
 // -----------------------------------------------------------------------------
 
 new Pool(
-    'PPV',
+    'PPV', // @phpstan-ignore-line PHPStan 1.10 reports a string-to-int conv here, seems like a bug
```
Comment (chrismiceli) [inline pinc/stages.inc:310]: is it worth setting up a minimal example and creating an issue against phpstan?

**Review by chrismiceli — APPROVED** [review]

**Review by srjfoo — APPROVED** [review]

File: pinc/stages.inc (hunk 3 above)
Comment (jchaffraix) [reply pinc/stages.inc:310]: Yes, I just did! Also the reproduction showed that the issue is that we reuse the name `Pool` that exists in the PHP standard library and confuses PHPStan:

https://github.com/phpstan/phpstan/issues/11303

Comment (chrismiceli) [reply pinc/stages.inc:310]: That's awesome, thanks.

**Pull request merged**